src/content/1.7/development/internationalization/translation/using-the-translator.md

```diff
@@ -0,0 +1,118 @@
+---
+menuTitle: Using the Translator
+title: Using the Translator to translate wordings
+weight: 30
+---
+
+# Using the Translator to translate wordings
+
+{{% notice note %}}
+This section provides an quick reference on how to use the Translator. For more information, read Symfony's documentation on [Using the Translator](https://symfony.com/doc/3.4/components/translation/usage.html).
+{{% /notice %}}
+
+## PHP files
+
+To translate wordings in PHP files, you first need an instance of the `Translator` service (explained below). Then, you can use the `trans()` method to translate your wording:
+
+```php
+<?php
+echo $translator->trans('This product is no longer available.', [], 'Shop.Notifications.Error');
+``` 
+
+The `trans()` method takes three arguments:
+
+1. `$id` – The wording you want to translate. Keep in mind that it has to be _exactly_ the same as the one in the default catalogue, or the translation won't work.
+2. `$parameters` – An array of replacements, if any. ([Learn more about translation placeholders](https://symfony.com/doc/3.4/components/translation/usage.html#component-translation-placeholders)).
+3. `$domain` – The [translation domain][translation-domains] for that wording.
+
+{{% notice warning %}}
+Be aware that in Symfony-based Admin controllers, the second and third arguments have been swapped in order to allow `$replacements` to be optional. For more, see [FrameworkBundleAdminController](https://github.com/PrestaShop/PrestaShop/blob/1.7.6.0/src/PrestaShopBundle/Controller/Admin/FrameworkBundleAdminController.php#L275).
+{{% /notice %}}
+
+### Inside controllers
+
+Controllers include a helper method named `trans()` that calls the translator internally:
+
+```php
+<?php
+// legacy controllers
+$this->trans('This product is no longer available.', [], 'Shop.Notifications.Error');
+
+// Symfony-based controllers (FrameworkBundleAdminController)
+$this->trans('This product is no longer available.', 'Shop.Notifications.Error', []);
+```
+
+### Outside controllers
+
+If you are outside a controller, and after careful consideration you think you absolutely need some stuff translated, then you can add it as a dependency of your class:
+
+```php
+<?php
+// SomeService.php
+
+namespace PrestaShop\PrestaShop\Core\Foo\Bar;
```

Comment:
```suggestion
// SomeService.php

declare(strict_types=1);

namespace PrestaShop\PrestaShop\Core\Foo\Bar;
```

Reply:
Do you think this is relevant for this example?

Reply:
As users/readers copy/paste code, always put good practices in sample code.

Reply:
It makes the example longer to read and understand, it drives the focus away from the subject.

Reply:
As you want.